service/common/src/main/java/org/apache/polaris/service/exception/IcebergExceptionMapper.java

```diff
@@ -144,4 +112,56 @@ public static boolean containsAnyAccessDeniedHint(String message) {
   public static Collection<String> getAccessDeniedHints() {
     return ImmutableSet.copyOf(ACCESS_DENIED_HINTS);
   }
+
+  static int mapExceptionToResponseCode(RuntimeException rex) {
+    // Cloud exceptions
+    if (rex instanceof S3Exception
+        || rex instanceof AzureException
+        || rex instanceof StorageException) {
```

Comment:
Why bother with complex logic when we have a typed switch below?

Reply:
Do you mean the 1st or 2nd switch? This guards the `doesAnyThrowableContainAccessDeniedHint` check, preserving the original behavior. I'm hoping to limit any unintended consequences by scoping this down to cloud errors.

Reply:
I mean that the logic seems to be overly complex when boolean expressions are combined with type-based switches in the same method... but I do not feel too strongly about this... it's a "nit" comment :)

Reply:
nit: maybe refactor storage exception handling into a separate (private) method?